packages/Utils/src/get-max-width-height.ts

```diff
@@ -1,6 +1,20 @@
-export const getMax = (width: string, height: string = width): string => {
-  const widthValue = parseInt(width, 10)
-  const heightValue = parseInt(height, 10)
-  const diff = widthValue - heightValue
+import { SystemProp, Theme } from '@wttj/xstyled-styled-components'
+
+export const getMax = (
+  width: SystemProp<string | number, Theme>,
+  height: SystemProp<string | number, Theme> = width
+): SystemProp<string | number, Theme> => {
+  const typeToValueExtractor: Record<
+    string,
+    (value: SystemProp<string | number, Theme>) => number
+  > = {
+    string: (value: string) => parseInt(value, 10),
+    number: (value: number): number => value,
+  }
+  const widthNumber =
+    typeToValueExtractor[typeof width as keyof typeof typeToValueExtractor]?.(width) || 0
+  const heightNumber =
+    typeToValueExtractor[typeof width as keyof typeof typeToValueExtractor]?.(height) || 0
+  const diff = widthNumber - heightNumber
```

Comment:
i dont see the problem that has been solved here, could you clarify it ?

Reply:
the `width` and`height` arguments were typed as `string` but have been enriched after the upgrade of xstyled & styled-components resulting in type errors